File: leetcode/Easy/0232-implement-queue-using-stacks/0232-implement-queue-using-stacks.cpp
class MyQueue {
  stack<int> stack1;
  stack<int> stack2;

public:
  MyQueue() {}

  void push(int x) { stack1.push(x); }

  int pop() {
    if (stack2.empty()) {
      while (!stack1.empty()) {
        stack2.push(stack1.top());
        stack1.pop();
      }
    }
    if (stack2.empty())
      return -1;
    int front = stack2.top();
    stack2.pop();
    return front;
  }

  int peek() {
    if (stack2.empty()) {
      while (!stack1.empty()) {
        stack2.push(stack1.top());
        stack1.pop();
      }
    }
    if (stack2.empty())
      return -1;
    return stack2.top();
  }

  bool empty() { return stack1.empty() && stack2.empty(); }
};

/**
 * Your MyQueue object will be instantiated and called as such:
 * MyQueue* obj = new MyQueue();
 * obj->push(x);
 * int param_2 = obj->pop();
 * int param_3 = obj->peek();
 * bool param_4 = obj->empty();
 */

File: leetcode/Easy/0232-implement-queue-using-stacks/0232-implement-queue-using-stacks.py
class MyQueue:

    def __init__(self):
        self.push_stack = []
        self.pop_stack = []

    def push(self, x: int) -> None:
        self.push_stack.append(x)

    def pop(self) -> int:
        if not self.pop_stack:
            while self.push_stack:
                temp = self.push_stack.pop()
                self.pop_stack.append(temp)
        return self.pop_stack.pop()

    def peek(self) -> int:
        if not self.pop_stack:
            while self.push_stack:
                temp = self.push_stack.pop()
                self.pop_stack.append(temp)  
        return self.pop_stack[-1]

    def empty(self) -> bool:
        return False if self.push_stack or self.pop_stack else True


# Your MyQueue object will be instantiated and called as such:
# obj = MyQueue()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.peek()
# param_4 = obj.empty()